core/data/src/main/java/uk/gov/gchq/gaffer/data/generator/CsvGenerator.java

```diff
@@ -195,19 +201,45 @@ public void setCommaReplacement(final String commaReplacement) {
     public static class Builder {
         private LinkedHashMap<String, String> fields = new LinkedHashMap<>();
         private LinkedHashMap<String, String> constants = new LinkedHashMap<>();
+
+        private LinkedHashMap<String, String> propertiesFromSchema = new LinkedHashMap<>();
         private String commaReplacement = COMMA_REPLACEMENT_DEFAULT;
         private Boolean quoted;
 
+        /**
+         * Stores any additional properties of an {@link Element}.
+         *
+         * @param propertyHeadersFromSchema the name of the property headers to be added
+         * @return a new {@link CsvGenerator.Builder}
+         */
+        public Builder propertyHeadersFromSchema(final LinkedHashMap<String, String> propertyHeadersFromSchema) {
+            for (final String key: propertyHeadersFromSchema.keySet()) {
+                fields.put(key, key + ":" + propertyHeadersFromSchema.get(key));
+            }
+            return this;
+        }
+
+        /**
+         * Adds the main identifiers of an {@link Element}, ie VERTEX, GROUP, SOURCE, DEST to the fields LinkedHashMap.
+         * These identifoers are predifined in a {@link CsvFormat} e.g. {@link Neo4jFormat}
+         *
+         * @param identifiersFromFormat the name of the headers to be added
+         * @return a new {@link CsvGenerator.Builder}
+         */
+        public Builder identifiersFromFormat(final LinkedHashMap<String, String> identifiersFromFormat) {
+            this.fields = identifiersFromFormat;
+            return this;
+        }
+
         /**
          * Stores the group of an {@link Element}.
          *
          * @param columnHeader the group of the {@code Element}
          * @return a new {@link Builder}
          */
         public Builder group(final String columnHeader) {
-            fields.put(GROUP, columnHeader);
-            return this;
-        }
+                return identifier(IdentifierType.GROUP, columnHeader);
+            }
```

Comment:
This is still showing as wrong for me.

Reply:
The indenting is wrong, why is this not picked up by checkstyle?

Reply:
Checkstyle doesn't do indenting.

Reply:
Would be nice if something did this automatically

Reply:
Google code style can be done by spotless. This would replace checkstyle. It would require a major diff to bring all of the codebase up to spec though.

Reply:
Regardless for this PR, this indenting still needs to be fixed